internal/cfapi/cfapi.go

```diff
@@ -108,13 +109,17 @@ func (c *Client) Sign(ctx context.Context, req *SignRequest) (*SignResponse, err
 	}
 	defer resp.Body.Close()
 
+	rayID := resp.Header.Get("CF-Ray")
+
 	api := APIResponse{}
 	if err := json.NewDecoder(resp.Body).Decode(&api); err != nil {
 		return nil, err
 	}
 
 	if !api.Success {
-		return nil, &api.Errors[0]
+		err := &api.Errors[0]
+		err.RayID = rayID
```

Comment:
The other idea I had was to keep `APIError` unchanged and return a wrapped error instead.